src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -212,14 +217,17 @@ private UserDefinitionsNameResolver(INameResolver globalNameResolver, IEnumerabl
 
             public TexlFunctionSet Functions => _globalNameResolver.Functions;
 
+            public IEnumerable<KeyValuePair<DName, FormulaType>> DefinedTypes => _globalNameResolver.DefinedTypes;
+
             public bool SuggestUnqualifiedEnums => _globalNameResolver.SuggestUnqualifiedEnums;
 
             public bool Lookup(DName name, out NameLookupInfo nameInfo, NameLookupPreferences preferences = NameLookupPreferences.None)
             {
                 // lookup in the local scope i.e., function params & body and then look in global scope.
                 if (_args.TryGetValue(name, out var value))
                 {
-                    var type = value.TypeIdent.GetFormulaType()._type;
+                    // Fetch type from resolved argTypes
+                    var type = this._argTypes[value.ArgIndex];
```

Comment:
>= this._argTypes[value.ArgIndex]; [](http://example.com/codeflow?start=29&length=33)

This could throw if _args.Length > _argTypes.Length. Can we add a check for this in create?